rlclientlib/live_model.cc

```diff
@@ -38,11 +38,14 @@ namespace reinforcement_learning
     _initialized = other._initialized;
     return *this;
   }
-  int live_model::init(api_status* status) {
+
+  int live_model::init(api_status* status, trace_callback binding_trace) {
     if (_initialized)
       return error_code::success;
 
-    const auto err_code = _pimpl->init(status);
+    binding_tracer* tracer = (binding_trace != nullptr) ? new binding_tracer(binding_trace) : nullptr;
```

Comment:
> binding_tracer* tracer = (binding_trace != nullptr) ? new binding_tracer(binding_trace) : nullptr [](start = 3, length = 98)

This should be handled by the factory method.

Reply:
Factory method is taking an i_trace interface implementation instead of callback. I think we don't want to change the interface of the factory method.

---
In reply to: [252731503](https://github.com/VowpalWabbit/reinforcement_learning/pull/38#discussion_r252731503) [](ancestors = 252731503)

Reply:
To NY team: what do you think to refactor the factory method create(....) to use a void*, instead of putting i_trace* there? (May have a separate PR for this)

---
In reply to: [252731503](https://github.com/VowpalWabbit/reinforcement_learning/pull/38#discussion_r252731503) [](ancestors = 252731503)

Reply:
Can you explain why a void* should be used?

Reply:
Had a discussion with Rajan and Jacob - it won't be needed anymore. My previous thought was to enable passing a callback to i_trace implementation.

See new change in CreateLiveModel() in rl.net.live_model.cc. Jacob proposed cool stuff there.